### PR TITLE
Add ES|QL async query api docs

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit adds detailed API docs for the ES|QL async apis, as well as moving the narrative to the ES|QL specific REST docs. Additionally, a few top-level lists are reflowed to reflect that there are now multiple APIs.

relates #102455

### CODE REVIEW

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks Chris!

File: docs/reference/esql/esql-apis.asciidoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[[esql-apis]]
+== {esql} APIs
+
+The {es} Query Language ({esql}) provides a powerful way to filter, transform,
+and analyze data stored in {es}, and in the future in other runtimes. For an
+overview of {esql} and related tutorials, see <<esql>>.
+
+* <<esql-query-api>>
```
Comment (bpintea) [inline docs/reference/esql/esql-apis.asciidoc:8]: In the docs preview, this link is rendered in italics, not sure why.

File: docs/reference/esql/esql-async-query-api.asciidoc [hunk 2]
```diff
@@ -1,27 +1,19 @@
...
 
-The async query API lets you asynchronously execute a search request,
-monitor its progress, and retrieve results as they become available.
+The async query API lets you asynchronously execute a query request,
+monitor its progress, and retrieve results when they become available.
 
-Executing an <<esql,ES|QL ({es} query language)>> is commonly quite fast,
-however searches across large data sets or frozen data can take some time.
-To avoid long waits, run an async {esql} search.
+Queries initiated by this API may return results or not.
```
Comment (bpintea) [inline docs/reference/esql/esql-async-query-api.asciidoc:12]: Maybe "Just like the sync API, queries initiated..." -- not sure what is meant with this sentence, but can also stay :).

**Review by bpintea — APPROVED** [review]
LGTM.

File: docs/reference/esql/esql-async-query-api.asciidoc (hunk 2 above)
Comment (ChrisHegarty) [reply docs/reference/esql/esql-async-query-api.asciidoc:12]: I was trying to get across the point that results OR an id would be returned. This point is better clarified later in the spec. I'll just remove this sentence from here as it's a little confusing.

File: docs/reference/esql/esql-apis.asciidoc (hunk 1 above)
Comment (ChrisHegarty) [reply docs/reference/esql/esql-apis.asciidoc:8]: Seems to be caused by different levels of discrete heading in the doc titles. I updated the query-api doc to match that of others, and this seems to have resolved it.

**Pull request merged**